### PR TITLE
Fix for Checking and ignoring Nan results also converting half to float

### PR DESCRIPTION
**Pull request by ruchibahl18**
float:

### CODE REVIEW

File: chessllm/chess_utils.py [hunk 1]
```diff
@@ -1,7 +1,8 @@
 import io
 import re
 from datetime import datetime
-
+# import os
+# os.add_dll_directory(r"C:\Program Files\GTK3-Runtime Win64\bin") Uncomment this line if running on windows and GTK3 is installed
```
Comment (kgourgou) [inline chessllm/chess_utils.py:5]: Hmm, there must be a better way than commenting this in and out.

File: chessllm/elo.py [hunk 2]
```diff
@@ -25,18 +25,19 @@ def update_elo_ratings(chess_data):
...
         model1 = row["Model1"]
         model2 = row["Model2"]
         result = row["Result"]
-
+        
         model1_elo = elo_ratings[model1]
         model2_elo = elo_ratings[model2]
 
-        result = [float(x) for x in result.split("-")]
+        result = [result_to_float(x) for x in result.split("-")]
```
Comment (kgourgou) [inline chessllm/elo.py:38]: :+1:

File: chessllm/chess_utils.py (hunk 1 above)
Comment (ruchibahl18) [reply chessllm/chess_utils.py:5]: Agreed, maybe I can check which os we are running on and add the import . Also will add the GTK installation step in readme.md

Comment (ruchibahl18) [reply chessllm/chess_utils.py]: FIxed comment by adding import only if platform is Windows

**Comment by kgourgou** [comment]
Looks pretty good @ruchibahl18 . Did you mean to keep the output in the notebook? I see that there were some things didn't run in the notebook. 

Also cc @mlabonne

**Comment by ruchibahl18** [comment]
> Looks pretty good @ruchibahl18 . Did you mean to keep the output in the notebook? I see that there were some things didn't run in the notebook.
> 
> Also cc @mlabonne

@kgourgou Apologies got too excited to commit StockFish notebook. I ll clean it up and then commit again

**Comment by ruchibahl18** [comment]
@kgourgou  and @mlabonne 
Cleared Outputs and also updated readme.md. Kindly review and merge if looks good

**Review by kgourgou — APPROVED** [review]

**Pull request merged**